app/controllers/evidence_controller.rb

```diff
@@ -122,11 +122,14 @@ def destroy
           # 2. from the node evidence tab
           # 3. from the evidence show page itself (under node)
           # When using redirect_back in case 3, we find an evidence not found error,
-          # since the evidence does not exist anymore. That's why we check the 'Referer' here:
-          if request.headers['Referer'] == project_node_evidence_url(current_project, @node, @evidence)
+          # since the evidence does not exist anymore. That's why we check the referrer here.
+          referrer_path = URI(request.referrer).tap { |u| u.query = nil }.to_s
```

Comment:
can you explain this code?

Reply:
This deletes the query part of the URL. This is one line version of:
```
uri = URI(request.referrer)
uri.query = nil
uri.to_s
```

Reply:
I've read we can use: `URI(request.headers['Referer']).path`

Reply:
And if request.referrer is not present we may got an error here ?

Reply:
I can confirm this throws an error when the referer is missing.